tests/models/mistral/test_modeling_mistral.py

```diff
@@ -483,6 +483,31 @@ def test_model_7b_logits(self):
         backend_empty_cache(torch_device)
         gc.collect()
 
+    @slow
+    @require_flash_attn
+    @require_torch_sdpa
+    def test_model_7b_logits_long_with_sdpa_and_flash2(self):
+        input_ids = [1] + [306, 338] * 2048
+        model = MistralForCausalLM.from_pretrained(
+            "mistralai/Mistral-7B-v0.1", device_map="auto", attn_implementation="flash_attention_2"
+        )
+        input_ids = torch.tensor([input_ids]).to(model.model.embed_tokens.weight.device)
+        with torch.no_grad():
+            out = model(input_ids).logits.cpu()
+
+        input_ids = [1] + [306, 338] * 2048
+        model = MistralForCausalLM.from_pretrained(
+            "mistralai/Mistral-7B-v0.1", device_map="auto", attn_implementation="sdpa"
+        )
+        input_ids = torch.tensor([input_ids]).to(model.model.embed_tokens.weight.device)
+        with torch.no_grad():
+            out1 = model(input_ids).logits.cpu()
+        torch.testing.assert_close(out.mean(-1), out1.mean(-1), atol=1e-2, rtol=1e-2)
```

Comment:
let's make sure we test all logits not just the mean
```suggestion
        torch.testing.assert_close(out, out1, atol=1e-4, rtol=1e-4)
```

with this, the test is failing: 
```python 
>       torch.testing.assert_close(out, out1, atol=1e-4, rtol=1e-4)
E       AssertionError: Tensor-likes are not close!
E       
E       Mismatched elements: 90967735 / 131104000 (69.4%)
E       Greatest absolute difference: 0.328125 at index (0, 2310, 338) (up to 0.0001 allowed)
E       Greatest relative difference: 114689.0 at index (0, 1267, 4581) (up to 0.0001 allowed)
```